src/awkward/_connect/numba/builder.py

```diff
@@ -74,8 +74,16 @@ def box_ArrayBuilder(arraybuildertype, arraybuilderval, c):
     ArrayBuilder_obj = c.pyapi.unserialize(
         c.pyapi.serialize_object(ak.highlevel.ArrayBuilder)
     )
-    behavior_obj = c.pyapi.unserialize(
-        c.pyapi.serialize_object(arraybuildertype.behavior)
+    serializable2dict_obj = c.pyapi.unserialize(
+        c.pyapi.serialize_object(ak._connect.numba.arrayview.serializable2dict)
+    )
+    behavior2_obj = c.pyapi.unserialize(
+        c.pyapi.serialize_object(
+            ak._connect.numba.arrayview.dict2serializable(arraybuildertype.behavior)
+        )
+    )
+    behavior_obj = c.pyapi.call_function_objargs(
+        serializable2dict_obj, (behavior2_obj,)
```

Comment:
@jpivarski - Thanks for the reminder! This is indeed a correct way to serialize a dict :-)